File: src/billing-service/FeatureFlags.cs
// SPDX-FileCopyrightText: 2026 Cedric Moulard / Kraftr
// SPDX-License-Identifier: MIT

using OpenFeature;
using OpenFeature.Contrib.Providers.Flagd;

namespace BillingService;

public static class FeatureFlags
{
    public static async Task InitAsync()
    {
        var host = Environment.GetEnvironmentVariable("FLAGD_HOST") ?? "flagd";
        var portStr = Environment.GetEnvironmentVariable("FLAGD_PORT") ?? "8013";
        var port = int.Parse(portStr);

        var provider = new FlagdProvider(new Uri($"http://{host}:{port}"));
        await Api.Instance.SetProviderAsync(provider);
    }

    public static FeatureClient Client => Api.Instance.GetClient();
}

File: src/payment-service/feature_flags.py
# SPDX-FileCopyrightText: 2026 Cedric Moulard / Kraftr
# SPDX-License-Identifier: MIT

import os
import logging

from openfeature import api
from openfeature.contrib.provider.flagd import FlagdProvider

logger = logging.getLogger("feature_flags")


def init_flags():
    host = os.getenv("FLAGD_HOST", "flagd")
    port = int(os.getenv("FLAGD_PORT", "8013"))
    api.set_provider(FlagdProvider(host=host, port=port))
    logger.info("flagd_provider_registered", extra={"host": host, "port": port})


def client():
    return api.get_client()
